src/main/scala/mesosphere/mesos/AppSpecResourceMatcher.scala

```diff
@@ -0,0 +1,71 @@
+package mesosphere.mesos
+
+import mesosphere.marathon.core.task.{ Task }
+import mesosphere.marathon.state.RunSpec
+import mesosphere.mesos.protos.Resource
+import org.apache.mesos.Protos.Offer
+import org.slf4j.LoggerFactory
+
+object AppSpecResourceMatcher {
+  private[this] val log = LoggerFactory.getLogger(getClass)
+  import ResourceMatcher.{ ResourceMatch, ResourceSelector }
+
+  def matchResources(offer: Offer, runSpec: RunSpec, runningTasks: => Iterable[Task],
+    selector: ResourceSelector): Option[ResourceMatch] = {
+    lazy val tasks =
+      runningTasks.filter(_.launched.exists(_.runSpecVersion >= runSpec.versionInfo.lastConfigChangeVersion))
```

Comment:
don't like this logic here: the tasks should already be filtered before entering this func